### ISSUE TITLE
Pre-configure settings

### ISSUE THREAD

**Issue by gakimball** [zurb/panini#6]
I'd like to rework the API to pre-configure your layouts, pages, partials, etc. instead of re-running that logic in the stream every time.

I think it can look similar to what we did for [Supercollider](https://github.com/gakimball/supercollider#setup). Something like this:

```js
panini.config({
  layouts: '...',
  pages: '...'
});
```

So Handlebars is configured beforehand, and then the actual parsing function used in the stream is just rendering the Handlebars templates.

**Comment by lyckron** [comment]
+1

So regarding gulp usage, are we talking something like:
```javascript
panini.config(options);

gulp.task('panini', function() {
 return panini.init();
});
```
?

**Comment by gakimball** [comment]
We can probably just do `panini()` instead of `panini.init()`. It's simpler to understand. Supercollider does that currently but I don't love it, and it should be an easy change.

**Comment by lyckron** [comment]
Agree, but if straying from that pattern anyway, plus keeping the file structure strict anyway, and the aim is to simplify , then maybe the `config` method should just be a private constructor, so that it would be:

```javascript
panini(options);
``` 
Then in the `index.js`:
```javascript
// Create or require Panini class
  
module.exports = function(options) {
  return new Panini(options);
}
```

**Comment by Abhra1992** [comment]
This seems like a fair way to introduce the plugin to gulp.
Do you guys have kind of a roadmap about where you want to see panini when it's released/
I'd be happy to pitch in for implementing some of the basic functionality.

**Comment by gakimball** [comment]
Implemented via e8a8b878edb91b2441a6881a6f7b1215234f2a6b
